src/factories/node.ts

```diff
@@ -179,15 +179,20 @@ export async function nodeContainerFactory(node: RunGraphNode) {
 
   function getNodeCacheKey(nodeData: RunGraphNode): string {
     const endTime = nodeData.end_time ?? new Date()
-    const artifactCount = nodeData.artifacts?.length ?? 0
+    const artifactCacheKey = nodeData.artifacts?.map(artifact => {
+      if (artifact.type === 'progress') {
+        return `${artifact.id}-${artifact.data}`
+      }
+      return artifact.id
+    }).join('|')
     const values = [
       nodeData.state_type,
       endTime.getTime(),
-      artifactCount,
       layout.horizontal,
       layout.horizontalScaleMultiplier,
       config.styles.colorMode,
-      artifactCount > 0 && settings.disableArtifacts,
+      artifactCacheKey && settings.disableArtifacts,
+      artifactCacheKey,
```

Comment:
Probably not hurting anything but I think this is what we want here. 
```suggestion
      artifactCacheKey && settings.disableArtifacts,
```